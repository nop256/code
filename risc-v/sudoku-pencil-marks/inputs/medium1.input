......6......9.......42..159.2.6....3.....5.841.3....2.4.6........5...9......8.37
