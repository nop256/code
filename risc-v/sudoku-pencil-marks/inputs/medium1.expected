Please enter an 81-character board:

Testing pencil_marks with the following board:

+-----------------------+-----------------------+-----------------------+
| 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 |
| 4 5 6   4 5 6   4 5 6 | 4 5 6   4 5 6   4 5 6 |   6     4 5 6   4 5 6 |
| 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6   4 5 6   4 5 6 | 4 5 6     9     4 5 6 | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 | 1 2 3 |       |       | 1 2 3 | 1 2 3 |       |       |
| 4 5 6   4 5 6   4 5 6 |   4       2     4 5 6 | 4 5 6     1       5   |
| 7 8 9 | 7 8 9 | 7 8 9 |       |       | 7 8 9 | 7 8 9 |       |       |
+-----------------------+-----------------------+-----------------------+
|       | 1 2 3 |       | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
|   9     4 5 6     2   | 4 5 6     6     4 5 6 | 4 5 6   4 5 6   4 5 6 |
|       | 7 8 9 |       | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |       |
|   3     4 5 6   4 5 6 | 4 5 6   4 5 6   4 5 6 |   5     4 5 6     8   |
|       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       |
|   4       1     4 5 6 |   3     4 5 6   4 5 6 | 4 5 6   4 5 6     2   |
|       |       | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       |
+-----------------------+-----------------------+-----------------------+
| 1 2 3 |       | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |
| 4 5 6     4     4 5 6 |   6     4 5 6   4 5 6 | 4 5 6   4 5 6   4 5 6 |
| 7 8 9 |       | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |
| 4 5 6   4 5 6   4 5 6 |   5     4 5 6   4 5 6 | 4 5 6     9     4 5 6 |
| 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 | 1 2 3 |       | 1 2 3 |       |       |
| 4 5 6   4 5 6   4 5 6 | 4 5 6   4 5 6     8   | 4 5 6     3       7   |
| 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 | 7 8 9 |       | 7 8 9 |       |       |
+-----------------------+-----------------------+-----------------------+

The return value is 1
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
| 1 2 . | . 2 3 | 1 . 3 | 1 . . | 1 . 3 | 1 . 3 |       | . 2 . | . . 3 |
| . 5 .   . 5 .   4 5 . | . . .   . 5 .   . 5 . |   6     4 . .   4 . . |
| 7 8 . | 7 8 9 | 7 8 9 | 7 8 . | 7 8 . | 7 . . |       | 7 8 . | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 3 | 1 . 3 | 1 . . |       | 1 . 3 | . 2 3 | . 2 . | . . 3 |
| . 5 6   . 5 6   4 5 6 | . . .     9     . 5 6 | 4 . .   4 . .   4 . . |
| 7 8 . | 7 8 . | 7 8 . | 7 8 . |       | 7 . . | 7 8 . | 7 8 . | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . | . . 3 | . . 3 |       |       | . . 3 | . . 3 |       |       |
| . . 6   . . 6   . . 6 |   4       2     . . 6 | . . .     1       5   |
| 7 8 . | 7 8 9 | 7 8 9 |       |       | 7 . . | 7 8 9 |       |       |
+-----------------------+-----------------------+-----------------------+
|       | . . . |       | 1 . . |       | 1 . . | 1 . 3 | . . . | 1 . 3 |
|   9     . 5 .     2   | . . .     6     4 5 . | 4 . .   4 . .   4 . . |
|       | 7 8 . |       | 7 8 . |       | 7 . . | 7 . . | 7 . . | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | . . . | . . . | 1 2 . | 1 . . | 1 2 . |       | . . . |       |
|   3     . . 6   . . 6 | . . .   4 . .   4 . . |   5     4 . 6     8   |
|       | 7 . . | 7 . . | 7 . 9 | 7 . . | 7 . 9 |       | 7 . . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | . . . |       | . . . | . . . | . . . | . . . |       |
|   4       1     . 5 6 |   3     . 5 .   . 5 . | . . .   . . 6     2   |
|       |       | 7 8 . |       | 7 8 . | 7 . 9 | 7 . 9 | 7 . . |       |
+-----------------------+-----------------------+-----------------------+
| 1 2 . |       | 1 . 3 |       | 1 . 3 | 1 2 3 | 1 2 . | . 2 . |       |
| . 5 .     4     . 5 . |   6     . . .   . . . | . . .   . 5 .     1   |
| 7 8 . |       | 7 8 9 |       | 7 . . | 7 . 9 | . 8 . | . 8 . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 3 | 1 . 3 |       | 1 . 3 | 1 2 3 | 1 2 . |       | 1 . . |
| . . 6   . . 6   . . 6 |   5     4 . .   4 . . | 4 . .     9     4 . 6 |
| 7 8 . | 7 8 . | 7 8 . |       | 7 . . | 7 . . | . 8 . |       | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 . | 1 . . | 1 2 . | 1 . . |       | 1 2 . |       |       |
| . 5 6   . 5 6   . 5 6 | . . .   4 . .     8   | 4 . .     3       7   |
| . . . | . . 9 | . . 9 | . . 9 | . . . |       | . . . |       |       |
+-----------------------+-----------------------+-----------------------+

The return value is 1
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
| 1 2 . | . 2 3 | 1 . 3 | 1 . . | 1 . 3 | 1 . 3 |       | . 2 . | . . 3 |
| . 5 .   . 5 .   4 5 . | . . .   . 5 .   . 5 . |   6     4 . .   4 . . |
| 7 8 . | 7 8 9 | 7 8 9 | 7 8 . | 7 8 . | 7 . . |       | 7 8 . | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 3 | 1 . 3 | 1 . . |       | 1 . 3 | . 2 3 | . 2 . | . . 3 |
| . 5 6   . 5 6   4 5 6 | . . .     9     . 5 6 | 4 . .   4 . .   4 . . |
| 7 8 . | 7 8 . | 7 8 . | 7 8 . |       | 7 . . | 7 8 . | 7 8 . | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . | . . 3 | . . 3 |       |       | . . 3 | . . 3 |       |       |
| . . 6   . . 6   . . 6 |   4       2     . . 6 | . . .     1       5   |
| 7 8 . | 7 8 9 | 7 8 9 |       |       | 7 . . | 7 8 9 |       |       |
+-----------------------+-----------------------+-----------------------+
|       | . . . |       | 1 . . |       | 1 . . | 1 . 3 | . . . | . . 3 |
|   9     . 5 .     2   | . . .     6     4 5 . | 4 . .   4 . .   4 . . |
|       | 7 8 . |       | 7 8 . |       | 7 . . | 7 . . | 7 . . | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | . . . | . . . | 1 2 . | 1 . . | 1 2 . |       | . . . |       |
|   3     . . 6   . . 6 | . . .   4 . .   4 . . |   5     4 . 6     8   |
|       | 7 . . | 7 . . | 7 . 9 | 7 . . | 7 . 9 |       | 7 . . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | . . . |       | . . . | . . . | . . . | . . . |       |
|   4       1     . 5 6 |   3     . 5 .   . 5 . | . . .   . . 6     2   |
|       |       | 7 8 . |       | 7 8 . | 7 . 9 | 7 . 9 | 7 . . |       |
+-----------------------+-----------------------+-----------------------+
| . 2 . |       | . . 3 |       | . . 3 | . 2 3 | . 2 . | . 2 . |       |
| . 5 .     4     . 5 . |   6     . . .   . . . | . . .   . 5 .     1   |
| 7 8 . |       | 7 8 9 |       | 7 . . | 7 . 9 | . 8 . | . 8 . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 3 | 1 . 3 |       | 1 . 3 | 1 2 3 | . 2 . |       | . . . |
| . . 6   . . 6   . . 6 |   5     4 . .   4 . . | 4 . .     9     4 . 6 |
| 7 8 . | 7 8 . | 7 8 . |       | 7 . . | 7 . . | . 8 . |       | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 . | 1 . . | 1 2 . | 1 . . |       | . 2 . |       |       |
| . 5 6   . 5 6   . 5 6 | . . .   4 . .     8   | 4 . .     3       7   |
| . . . | . . 9 | . . 9 | . . 9 | . . . |       | . . . |       |       |
+-----------------------+-----------------------+-----------------------+

The return value is 0
After pencil_marks returned the board is now:

+-----------------------+-----------------------+-----------------------+
| 1 2 . | . 2 3 | 1 . 3 | 1 . . | 1 . 3 | 1 . 3 |       | . 2 . | . . 3 |
| . 5 .   . 5 .   4 5 . | . . .   . 5 .   . 5 . |   6     4 . .   4 . . |
| 7 8 . | 7 8 9 | 7 8 9 | 7 8 . | 7 8 . | 7 . . |       | 7 8 . | . . 9 |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 3 | 1 . 3 | 1 . . |       | 1 . 3 | . 2 3 | . 2 . | . . 3 |
| . 5 6   . 5 6   4 5 6 | . . .     9     . 5 6 | 4 . .   4 . .   4 . . |
| 7 8 . | 7 8 . | 7 8 . | 7 8 . |       | 7 . . | 7 8 . | 7 8 . | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| . . . | . . 3 | . . 3 |       |       | . . 3 | . . 3 |       |       |
| . . 6   . . 6   . . 6 |   4       2     . . 6 | . . .     1       5   |
| 7 8 . | 7 8 9 | 7 8 9 |       |       | 7 . . | 7 8 9 |       |       |
+-----------------------+-----------------------+-----------------------+
|       | . . . |       | 1 . . |       | 1 . . | 1 . 3 | . . . | . . 3 |
|   9     . 5 .     2   | . . .     6     4 5 . | 4 . .   4 . .   4 . . |
|       | 7 8 . |       | 7 8 . |       | 7 . . | 7 . . | 7 . . | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       | . . . | . . . | 1 2 . | 1 . . | 1 2 . |       | . . . |       |
|   3     . . 6   . . 6 | . . .   4 . .   4 . . |   5     4 . 6     8   |
|       | 7 . . | 7 . . | 7 . 9 | 7 . . | 7 . 9 |       | 7 . . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
|       |       | . . . |       | . . . | . . . | . . . | . . . |       |
|   4       1     . 5 6 |   3     . 5 .   . 5 . | . . .   . . 6     2   |
|       |       | 7 8 . |       | 7 8 . | 7 . 9 | 7 . 9 | 7 . . |       |
+-----------------------+-----------------------+-----------------------+
| . 2 . |       | . . 3 |       | . . 3 | . 2 3 | . 2 . | . 2 . |       |
| . 5 .     4     . 5 . |   6     . . .   . . . | . . .   . 5 .     1   |
| 7 8 . |       | 7 8 9 |       | 7 . . | 7 . 9 | . 8 . | . 8 . |       |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 3 | 1 . 3 |       | 1 . 3 | 1 2 3 | . 2 . |       | . . . |
| . . 6   . . 6   . . 6 |   5     4 . .   4 . . | 4 . .     9     4 . 6 |
| 7 8 . | 7 8 . | 7 8 . |       | 7 . . | 7 . . | . 8 . |       | . . . |
| - - - + - - - + - - - | - - - + - - - + - - - | - - - + - - - + - - - |
| 1 2 . | . 2 . | 1 . . | 1 2 . | 1 . . |       | . 2 . |       |       |
| . 5 6   . 5 6   . 5 6 | . . .   4 . .     8   | 4 . .     3       7   |
| . . . | . . 9 | . . 9 | . . 9 | . . . |       | . . . |       |       |
+-----------------------+-----------------------+-----------------------+
